code/modules/fishing/fishing.dm

```diff
@@ -0,0 +1,202 @@
+/*
+ * Fishing! Contains normal fishing rods and nets.
+ */
+
+GLOBAL_LIST_INIT(generic_fishing_rare_list, list(
+		/mob/living/simple_mob/animal/passive/fish/solarfish = 1,
+		/mob/living/simple_mob/animal/passive/fish/icebass = 5,
+		/mob/living/simple_mob/animal/passive/fish/koi = 3
+		))
+
+GLOBAL_LIST_INIT(generic_fishing_uncommon_list, list(
+		/mob/living/simple_mob/animal/passive/fish/salmon = 6,
+		/mob/living/simple_mob/animal/passive/fish/pike = 10,
+		/mob/living/simple_mob/animal/passive/fish/javelin = 3,
+		/mob/living/simple_mob/animal/passive/crab/sif = 1
+		))
+
+GLOBAL_LIST_INIT(generic_fishing_common_list, list(
+		/mob/living/simple_mob/animal/passive/fish/bass = 10,
+		/mob/living/simple_mob/animal/passive/fish/trout = 8,
+		/mob/living/simple_mob/animal/passive/fish/perch = 6,
+		/mob/living/simple_mob/animal/passive/fish/murkin = 8,
+		/mob/living/simple_mob/animal/passive/fish/rockfish = 5,
+		/mob/living/simple_mob/animal/passive/crab = 1
+		))
+
+GLOBAL_LIST_INIT(generic_fishing_junk_list, list(
+		/obj/item/clothing/shoes/boots/cowboy = 1,
+		/obj/random/fishing_junk = 10
+		))
+
+GLOBAL_LIST_INIT(generic_fishing_pool_list, list(
+		/obj/item/weapon/bikehorn/rubberducky = 5,
+		/obj/item/toy/plushie/carp = 20,
+		/obj/random/junk = 80,
+		/obj/random/trash = 80,
+		/obj/item/weapon/spacecash/c1 = 10,
+		/obj/item/weapon/spacecash/c10 = 5,
+		/obj/item/weapon/spacecash/c100 = 1
+		))
+
+#define FISHING_RARE     "rare"
+#define FISHING_UNCOMMON "uncommon"
+#define FISHING_COMMON   "common"
+#define FISHING_JUNK     "junk"
+#define FISHING_NOTHING  "nothing"
+
+GLOBAL_LIST_INIT(generic_fishing_chance_list, list(FISHING_RARE = 5, FISHING_UNCOMMON = 15, FISHING_COMMON = 30, FISHING_JUNK = 30, FISHING_NOTHING = 40))
+
+/turf/simulated/floor/water
+	var/has_fish = TRUE //If the water has fish or not.
+
+	var/list/rare_fish_list	// Rare list.
+
+	var/list/uncommon_fish_list	// Uncommon list.
+
+	var/list/common_fish_list	// Common list.
+
+	var/list/junk_list	// Junk item list.
+
+	var/list/fishing_loot	// Chance list.
+
+	var/fishing_cooldown = 30 SECONDS
+	var/last_fished = 0
+
+	var/fish_type
+	var/min_fishing_time = 30	// Time in seconds.
+	var/max_fishing_time = 90
+
+	var/being_fished = FALSE
+
+/turf/simulated/floor/water/proc/handle_fish()	// Subtypes should over-ride this, and supply their own GLOB lists for maximum Mix and Match power.
+	if(has_fish)
+		rare_fish_list = GLOB.generic_fishing_rare_list
+		uncommon_fish_list = GLOB.generic_fishing_uncommon_list
+		common_fish_list = GLOB.generic_fishing_common_list
+		junk_list = GLOB.generic_fishing_junk_list
+		fishing_loot = GLOB.generic_fishing_chance_list
```

Comment:
Lines 5 - 40

Reply:
Oh, I was just blind and didn't see it because it's a single-line list declaration instead of a multi-line declaration. Was specifically asking about `generic_fishing_chance_list`